torch/distributed/rpc/api.py

```diff
@@ -136,6 +136,16 @@ def _set_proceed_shutdown_signal(sequence_id):
     ), "Termination signal sequence id {} got set twice.".format(sequence_id)
     proceed_signal.set()
 
+thread_local_var = threading.local()
+
+@contextlib.contextmanager
+def wait_all():
```

Comment:
For public APIs, we will need to add API docs for it. You can refer to other public APIs in this file (e.g., `shutdown`) as examples. The rString in `shutdown()` API compiles to this API doc: https://pytorch.org/docs/master/rpc.html#torch.distributed.rpc.shutdown

Reply:
To build docs locally and verify results, you can follow this steps https://github.com/pytorch/pytorch#building-the-documentation. It might be a little hard to do it in fbcode (but it's possible, let me know if you want to try that path). You can, however,  clone PyTorch repo to your laptop, and build PyTorch and the html doc files locally, and then use [this script](https://gist.github.com/colesbury/df218a96334dbd31ee0e93760f70f114) to serve it.

Reply:
<img width="1413" alt="wait_all_documentation" src="https://user-images.githubusercontent.com/68608730/89812093-9808e900-db0d-11ea-8c00-3891fc26c3e2.png">